### DISCUSSION TITLE
Could not boot into a newly recompiled kernel

### DISCUSSION THREAD

**Discussion by toncho11** [microsoft/WSL#8924] [General]
### Version

version 10.0.19042.1706

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64

### Distro Version

Ubuntu 20.04 LTS

### Repro Steps

I need to enable Bluetooth in the kernel. 
I did  recompile the kernel following [this tutorial](https://github.com/Diegorro98/usbipd-win-wiki/blob/405936ec3cfa2497d7bbc8ab170d4aa063e9442e/WSL-support.md).

I am using this config and kernel file:
[kernel_and_config.zip](https://github.com/microsoft/WSL/files/9672723/kernel_and_config.zip)

I have the impression that the config file works because if I give a wrong path it reports a file not found error. 

### Expected Behavior

The new kernel to boot.

### Actual Behavior

It takes 20 seconds and then the message "Le délai de l’opération a expiré, car aucune réponse n’a été reçue de l’ordinateur virtuel ou du conteneur." It is a timed out message that Ubuntu could not start (when the new kernel is configured).

**Comment by benhillis** [comment]
I'd suggest comparing your kernel to the default config, I suspect you may be missing vsock support.

**Comment by toncho11** [comment]
So there are different .vhdx files in WSL2 per Windows user, right? The problem I think is that you never know which image you use - is it different when you start WSL2 with administrative rights or when you log in as an administrator. Not knowing this in the beginning creates a great confusion. 

There should be a text file created in each image saying which is the .vhdx image (or Windows user) used. The confusion comes because you have the same Linux users, but different image depending on the Windows user, right?
